src/test/ui/error-codes/E0401.rs

```diff
@@ -8,11 +8,28 @@
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
 
+trait Baz<T> {}
+
 fn foo<T>(x: T) {
-    fn bar(y: T) { //~ ERROR E0401
+    fn bar<U, V: Baz<U>, W: Fn()>(y: T) { //~ ERROR E0401
```

Comment:
When you get back to this, add another test case like this one where the type arguments are all on their own line. I wan't to make sure the suggestion presentation will work will in the case or multiline spans. If the output looks wonky through no fault of your code, let's just file a ticket for it, as it would very likely already affect other diagnostics, and I don't want to block this PR.

Reply:
It doesn't work really well:
```rust
// Source
    fn baz<U,
           V: Baz<U>,
           W: Fn()>
           (y: T) {
    }

// Message
error[E0401]: can't use type parameters from outer function
  --> main.rs:19:16
   |
13 | fn foo<T>(x: T) {
   |        - type variable from outer function
...
19 |            (y: T) {
   |                ^ use of type variable from outer function
help: try using a local type parameter instead
   |
16 |     fn baz<U,
17 |            V: Baz<U>,
18 |            W: Fn()>
19 |            <T>(y: T) {
   |
```

When I implemented `span_extend_to_prev_char()` from `span_until_char()`, I removed the condition `!prev_source.contains('\n')`. I suspected that it could pose some problems but I didn't test it (I forgot I think...). Anyway if I update `span_extend_to_prev_char()` to not extend the span if a newline is present in the extension, I can have a message without the suggested snippet and just the text:
```
error[E0401]: can't use type parameters from outer function
  --> main.rs:19:16
   |
13 | fn foo<T>(x: T) {
   |        - type variable from outer function
...
19 |            (y: T) { //~ ERROR E0401
   |                ^ use of type variable from outer function
   |
   = help: try using a local type parameter instead
```

I think this is better since it doesn't change the message for the other cases and makes this one clearer. What do you think, should I push this patch?

Reply:
Push the latter change to avoid giving an incorrect suggestion, but could you also file ticket once the PR is merged to account for this? As rare as the case might be, it'd be nice to eventually support it too. Add the change to the test so that we can verify that the suggestion is not being displayed for this case.